rectools/visuals/visual_app.py

```diff
@@ -0,0 +1,273 @@
+import typing as tp
+
+import ipywidgets as widgets
+import numpy as np
+import pandas as pd
+from IPython.display import display
+
+from rectools import Columns
+
+
+class AppDataStorage:
+    """
+    Storage and processing of data for `VisualApp` widgets. This class is not meant to be used
+    directly. Use `VisualApp` class instead
+    """
+
+    def __init__(
+        self,
+        recos: tp.Dict[tp.Hashable, pd.DataFrame],
+        item_data: pd.DataFrame,
+        requests_dict: tp.Dict[tp.Hashable, tp.Hashable],
+        is_u2i: bool = True,
```

Comment:
I see your point. For example I can think of simple visualisation of predefined item lists (like hand-made recos and popular recos for cold scenarios) without the need for interactions history at all. 
But if we think that this is going to be an option, should we move all of the use cases to different classes? Because each option has very strict demands for the input data. And it is going to be very confusing to try to figure out necessary parameters for each case.
Like: pass interactions for u2i but don't pass for i2i and cold. And so on

Reply:
Yes, probably having a separate class for each case is a good idea

Reply:
Check it out. Not sure about ABC and abstract method decorator here. Should we add it? Also I decided to change AppDataStorage to a private class after all. Probably VisualAppBase should also be private?

Reply:
Don't use ABC and decorators please, use `NotImplemented` / `NotImplementedError` instead if you really need it.

I really don't see any reason to make `VisualAppBase` private. We have base classes for models, metrics, splitters and all of them are public.

Reply:
Ok. This thread is ready for review. Let's get the final version of VisualApp classes in this PR. And work on final version of DataStorage in the next PR